Exclude deprecated properties in count for case property warning

gherceg opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
https://dimagi.atlassian.net/browse/SAAS-18051

Review by commit 🐟 

The initial ask is pretty straightforward. When viewing the case config page or data dictionary, we show a warning to users if the number of case properties for the current case type exceeds a threshold. Currently this number includes deprecated case properties, but we want to exclude them from this count.

This was very straightforward for the case config page since we just pull the count of case properties specifically for this warning, and can change that function to exclude deprecated properties without consequence.

The data dictionary on the other hand relied on the total case property count for other things like pagination. It might be overkill, but I decided to return the total count, and the deprecated count, leaving the frontend to subtract off that number to determine the number of non-deprecated case properties. This felt much clearer from a code readability perspective.

I then wanted to add tests for my new deprecated property count and found the data dictionary tests hard to edit. The helper to assert the response for every test makes it so that we have to enable things for all tests (like the CASE_IMPORT_DATA_DICTIONARY_VALIDATION flag), and I find generally makes it challenging to add new tests easily. I hope the changes I made are an improvement, but let me know if you disagree. I think I may have added some fairly useless tests, but was trying to account for the fact that the tests were certainly thorough before given they asserted the entire response.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Pretty minor change that results in the worst case, displaying a warning when we shouldn't (or not displaying it when we should). That doesn't block anyone so the blast radius should be pretty small.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
Added tests to ensure deprecated case property counts are valid for the data dictionary.

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Rename properties_count to property_count

gherceg committed:
Pass deprecatedPropertyCount into data dictionary

gherceg committed:
Only include non-deprecated properties in count

gherceg committed:
Remove teardowns and addClassCleanup where needed

gherceg committed:
Move case type endpoint tests into own class

gherceg committed:
Isolate case properties endpoint tests

gherceg committed:
Sort imports

gherceg commented on corehq/apps/data_dictionary/tests/test_views.py:
```diff
@@ -29,15 +29,12 @@ class DataDictionaryViewTestBase(TestCase):
+
         cls.user = WebUser.create(None, 'username', 'Passw0rd!', None, None)
         cls.user.add_domain_membership(cls.domain_name, is_admin=True)
         cls.user.save()
```
Yeah specifically for tests 😢 https://github.com/dimagi/commcare-hq/blob/d918dbb21bfa1a4e91034d580bd6add5ceded7ee/corehq/apps/domain/deletion.py#L153-L154.

I don't really want to rely on that for cleaning up the web users, which is why I still included that cleanup as well.

gherceg commented on corehq/apps/data_dictionary/tests/test_views.py:
```diff
@@ -357,7 +356,121 @@ def test_restore_case_type(self):
 @es_test(requires=[case_search_adapter], setup_class=True)
-@patch('corehq.apps.data_dictionary.views.get_case_type_app_module_count', return_value={})
+@privilege_enabled(privileges.DATA_DICTIONARY)
+class CaseTypesTest(DataDictionaryViewTestBase):
```
Ah yeah I think I could clean that up. I was tempted to change the actual endpoint function names, but should probably just match the endpoint as is for now. Will make that change.

gherceg replied:
Didn't preserve the exact test name since I effectively split it out into two test classes, but they now correspond to the specific endpoints/functions: 657942582a20ee2955739f6fb2d485dfdcd7130f

gherceg committed:
Test names should match endpoint/function

mkangia committed:
fix filter

mkangia commented on corehq/apps/data_dictionary/util.py:
```diff
@@ -429,4 +429,4 @@ def get_case_property_count(domain, case_type):
         return 0
 
-    return case_type.properties.filter(property__deprecated=False).count()
+    return case_type.properties.filter(deprecated=False).count()
```
Hey @millerdev 
I will defer to @gherceg to respond to that. I guess a test is missing.

QA noted a 500 when testing a different PR on staging, so I looked into it and found the issue.

gherceg replied:
Thanks for resolving! I think this function is untested so I can add some simple tests for it.

gherceg committed:
Add tests for get_case_property_count

gherceg committed:
Remove unnecessary teardowns

SQL objects are created within a transaction for each test which is
rolled back. If created in setUpCls, they are flushed at the end of the
test class run
